docs/Producing effective SARIF.md

```diff
@@ -393,13 +393,25 @@ Similarly, most 'result' objects contain at least one 'artifactLocation' object.
 
 #### Messages
 
+##### `AvoidDuplicativeAnalysisTarget`: warning
+
+The 'analysisTarget' property '{1}' at '{0}' is unnecessary because it is the same as the result location. Remove the 'analysisTarget' property.
```

Comment:
>unnecessary [](start = 48, length = 11)

Please add something like: 'The 'analysisTarget' property is used to distinguish cases when a result fires in a file (such as an included header) that is different than the file that was scanned (such as a .cpp file that included the header). This C++ example example is the canonical scenario that led to this design pattern. 

Also, though it is implied, the reason we recommend removing it is to optimize file size.

The 'analysisTarget' property '{1}' at '{0}' can be removed because it is the same as the result location. This unnecessarily increases log file size. 'The 'analysisTarget' property is used to distinguish cases when a result fires in a file (such as an included header) that is different than the file that was scanned (such as a .cpp file that included the header).

Reply:
@eddynaka, +1 on Michael's rephrasing (the very last paragraph in his comment above).

---
In reply to: [455904829](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455904829) [](ancestors = 455904829)

Reply:
... except that instead of "when a result fires" I suggest "when a tool detects a result".

---
In reply to: [455907085](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455907085) [](ancestors = 455907085,455904829)

Reply:
I didn't intend to take over reauthoring the words, just point you in the right direction. :) feel free to send Eddy updates after it goes through your wordsmithing process. Every string needs a pass in my estimation. I need the team to send me the logs you generated in testing so that I can see what your new output looks like in bulk. If we aren't yet doing that testing, it needs to be included in the formalized process.

---
In reply to: [455907934](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455907934) [](ancestors = 455907934,455907085,455904829)

Reply:
just updated this as well!

---
In reply to: [455913218](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455913218) [](ancestors = 455913218,455907934,455907085,455904829)